KsApi/Service+DecodeHelpers.swift

```diff
@@ -22,25 +22,18 @@ extension Service {
       }
   }
 
-  func decodeModels<T: Decodable>(_ jsonData: Data) -> SignalProducer<[T], ErrorEnvelope> {
-    return SignalProducer(value: jsonData)
-      .flatMap { data -> SignalProducer<[T], ErrorEnvelope> in
-        do {
-          let decodedObject = try JSONDecoder().decode([T].self, from: data)
-
-          print("🔵 [KsApi] Successfully Decoded Data")
-
-          return .init(value: decodedObject)
-        } catch {
-          print("🔴 [KsApi] Failure - Decoding error: \(error)")
-          return .init(error: .couldNotDecodeJSON(error))
-        }
-      }
-  }
-
   func decodeModel<T: Decodable>(data json: Data) ->
     SignalProducer<T?, ErrorEnvelope> {
     return SignalProducer(value: json)
       .map { json in try? JSONDecoder().decode(T.self, from: json) }
   }
+
+  func decodeModel<T: Decodable>(data json: Data, ofType type: T.Type) -> Result<T?, ErrorEnvelope> {
```

Comment:
optional: I wonder if we should name these `decodeModelToResult` and/or `decodeModelToSignalProducer`. It'd add some extra overhead, especially if we switch all current usages (or replace all future usages of `decodeModelToResult` once that's the only one in use), but it would add a little clarity.

Reply:
Oh, I like that suggestion. Lemmee throw that in the auto-refactorer.